File: app/src/main/java/com/arturogutierrez/openticator/executor/MainThread.java
package com.arturogutierrez.openticator.executor;

import javax.inject.Inject;
import rx.Scheduler;
import rx.android.schedulers.AndroidSchedulers;

public class MainThread implements PostExecutionThread {

  @Inject
  public MainThread() {
    // Empty
  }

  @Override
  public Scheduler getScheduler() {
    return AndroidSchedulers.mainThread();
  }
}

File: app/src/main/java/com/arturogutierrez/openticator/executor/MainThread.kt
package com.arturogutierrez.openticator.executor

import rx.Scheduler
import rx.android.schedulers.AndroidSchedulers
import javax.inject.Inject

class MainThread @Inject constructor() : PostExecutionThread {
  override val scheduler: Scheduler
    get() = AndroidSchedulers.mainThread()
}
